### ISSUE TITLE
Compact output format, finer control over the output

### ISSUE THREAD

**Issue by lucc** [sharkdp/hyperfine#70]
I am running hyperfine with many different commands at once. In order to not only see the factor between the different commands at the end but also (some) of the means I suggest one, some or all of these changes to the output (switchable by a command line option):
* Do not print empty lines between mean and min-max and the headlines and so on. I can currently get quite close to what I want when I run `hyperfine ... | sed '/^ *$/d'`.
* The above mentioned  `hyperfine ... | sed '/^ *$/d'` is without color and when I run  `hyperfine -s full ... | sed '/^ *$/d'` instead I also get the interactive graph which doesn't play nicely with the pipe and sometimes messes up a line. So why not add a `color` style to `-s` which has color but is not interactive?
* All of these could be an additional value for `-s` and `-s` could be a comma separated list so that things like this would make sense: `hyperfine -s compact,color`, `hyperfine -s interactive,compact,nocolor`, `hyperfine -s full,compact`, `hyperfne -s basic,compact`...

**Comment by sharkdp** [comment]
Thank you for the feedback!

> Do not print empty lines between mean and min-max and the headlines and so on. I can currently get quite close to what I want when I run hyperfine ... | sed '/^ *$/d'.

I agree, we could save up some vertical space here. What do you think of the following compromise: skip all blank lines, except between two benchmarks:

```
Benchmark #1: sleep 0.1
  Time (mean ± σ):     101.1 ms ±   0.1 ms    [User: 0.7 ms, System: 1.2 ms]
  Range (min … max):   100.5 ms … 101.3 ms

Benchmark #2: sleep 0.01
  Time (mean ± σ):      11.1 ms ±   0.1 ms    [User: 0.9 ms, System: 0.6 ms]
  Range (min … max):    10.8 ms …  11.4 ms

Benchmark #3: sleep 0.11
  Time (mean ± σ):     111.2 ms ±   0.1 ms    [User: 1.1 ms, System: 0.4 ms]
  Range (min … max):   111.1 ms … 111.3 ms

Summary
  'sleep 0.01' ran
    9.09x faster than 'sleep 0.1'
    9.99x faster than 'sleep 0.11'
```

> The above mentioned hyperfine ... | sed '/^ *$/d' is without color and when I run hyperfine -s full ... | sed '/^ *$/d' instead I also get the interactive graph which doesn't play nicely with the pipe and sometimes messes up a line. So why not add a color style to -s which has color but is not interactive?

Sounds good to me!

> All of these could be an additional value for -s and -s could be a comma separated list so that things like this would make sense: hyperfine -s compact,color, hyperfine -s interactive,compact,nocolor, hyperfine -s full,compact, hyperfne -s basic,compact...

If we would strip some blank lines by defaultm, as suggested above, do you think we would still need a `compact` option?

**Comment by lucc** [comment]
I am fine with making the semi compact output that you defined the default. My suggestion for the "compact" keyword was only for backwards compatibility.

**Comment by sharkdp** [comment]
Ok, so let's:

- [ ] Strip the empty lines
- [ ] Add `color` as new type for `--style`.

**Comment by jasonpeacock** [comment]
Working on this...

**Comment by jasonpeacock** [comment]
PR submitted.

**Comment by sharkdp** [comment]
Released in [v1.3.0](https://github.com/sharkdp/hyperfine/releases/tag/v1.3.0).
